stl/inc/flat_set

```diff
@@ -555,42 +561,55 @@ private:
     }
 
     template <class _Ty>
-        requires _Keylt_transparent || is_same_v<_Ty, _Kty>
-    size_type _Erase(_Ty&& _Val) {
-        const auto [_First, _Last] = equal_range(_Val);
+    size_type _Erase(const _Ty& _Val) {
+        _STL_INTERNAL_STATIC_ASSERT(_Keylt_transparent || is_same_v<_Ty, _Kty>);
+
+        if constexpr (!_Multi && is_same_v<_Ty, _Kty>) {
+            const key_compare& _Compare = _Get_comp();
+            const iterator _Where       = lower_bound(_Val);
+            if (_Where != end() && !_Compare(_Val, *_Where)) {
+                _Get_cont().erase(_Where);
+                return 1;
+            }
+            return 0;
+        } else {
+            const auto [_First, _Last] = equal_range(_Val);
 
-        const difference_type _Removed = _STD distance(_First, _Last);
-        _Get_cont().erase(_First, _Last);
-        return _Removed;
+            const auto _Removed = static_cast<size_type>(_Last - _First);
+            _Get_cont().erase(_First, _Last);
+            return _Removed;
+        }
     }
 
-    template <class _Other>
-        requires _Keylt_transparent || is_same_v<_Other, _Kty>
-    _NODISCARD iterator _Find(const _Other& _Val) {
-        const iterator _End   = end();
-        const iterator _Where = lower_bound(_Val);
-        if (_Where != _End && _Keys_equal(*_Where, _Val)) {
+    template <class _Ty>
+    _NODISCARD iterator _Find(const _Ty& _Val) {
+        _STL_INTERNAL_STATIC_ASSERT(_Keylt_transparent || is_same_v<_Ty, _Kty>);
+
+        const key_compare& _Compare = _Get_comp();
+        const iterator _End         = end();
+        const iterator _Where       = lower_bound(_Val);
+        if (_Where != _End && !_Compare(_Val, *_Where)) {
             return _Where;
         } else {
             return _End;
         }
     }
 
-    template <class _Other>

```

Comment:
I think it's ok to rename to `_Ty` as the type includes `key_type`, and `_Emplace` uses `_Ty` too.